core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleManager.scala

```diff
@@ -40,7 +84,24 @@ private[spark] class SortShuffleManager(conf: SparkConf) extends ShuffleManager
       shuffleId: Int,
       numMaps: Int,
       dependency: ShuffleDependency[K, V, C]): ShuffleHandle = {
-    new BaseShuffleHandle(shuffleId, numMaps, dependency)
+    if (!dependency.mapSideCombine && SortShuffleWriter.shouldBypassMergeSort(
```

Comment:
Can you add some comment explaining why this only works when aggregator & mapsidecombine is off?